### PR TITLE
e2e: do not set platform to fix running on Mac

### PR DESCRIPTION
**Pull request by timoreimann**
Newer versions of Docker seem to fail when using the default platform since Darwin on amd64 has become a dedicated image architecture with Docker:

```
e2e_test.go:284: end-to-end tests failed: failed to create container: Error response from daemon: image with reference docker.io/digitalocean/k8s-e2e-test-runner:latest was found but does not match the specified platform: wanted darwin/amd64, actual: linux/amd64
```

Not setting the platform ensures the Linux image is requested.

### CODE REVIEW

File: test/e2e/docker_test.go [hunk 1]
```diff
@@ -51,7 +50,7 @@ type containerParams struct {
 // It returns an error when the container startup or cleanup fails, or when the
 // container returns a non-zero exit code.
 func runContainer(ctx context.Context, p containerParams) (retErr error) {
-	cli, err := client.NewEnvClient()
+	cli, err := client.NewClientWithOpts(client.FromEnv)
```
Comment (timoreimann) [inline test/e2e/docker_test.go:53]: Drive-by improvement because the old method is deprecated.

**Review by varshavaradarajan — APPROVED** [review]
